examples/table_large_grow.rs

```diff
@@ -0,0 +1,55 @@
+//! Example demonstrating WebAssembly table growth limits
+//!
+//! Tests growing a table with a very large delta (0xff_ff_ff_ff) which should fail
+//! gracefully by returning -1 according to the WebAssembly specification.
+//!
+//! You can run the example directly by executing:
+//! ```shell
+//! cargo run --example table-large-grow --release --features backend
+//! ```
+
+use wasmer::{imports, Instance, Module, Store};
+
+fn main() -> anyhow::Result<()> {
+    let module_wat = r#"
+    (module
+      (table $t0 0 externref)
+      (table $t1 10 externref)
+      (func $init (export "init") (result i32)
+	(if (i32.ne (table.size $t1) (i32.const 10))
+	  (then (unreachable))
+	)
+	;; Store the result instead of dropping it
+	(table.grow $t0 (ref.null extern) (i32.const 0xff_ff_ff_ff))
+      )
+    )
+    "#;
+
+    let mut store = Store::default();
+    let module = Module::new(&store, &module_wat)?;
+    let import_object = imports! {};
+    let instance = Instance::new(&mut store, &module, &import_object)?;
+
+    let init = instance.exports.get_function("init")?;
+    let result = init.call(&mut store, &[])?;
+    #[cfg(target_os = "macos")]
+    assert_eq!(
+        result[0],
+        wasmer::Value::I32(0),
+        "On macOS, table.grow should return 0"
+    );
+
+    #[cfg(target_os = "linux")]
+    assert_eq!(
+        result[0],
+        wasmer::Value::I32(-1),
+        "On Linux, table.grow should return -1"
+    );
```

Comment:
This shouldn't happen (different output for same program, under different OSs). Good catch.
We need to make the API deterministic no matter what OS is running on

Reply:
Not sure/haven't put more thought into whether this is possible.

Pasting the scenarios here (before this PRs changes), since I have them handy, for some food for thought.

Linux:
* Linux with `vm.overcommit_memory = 0` results in `memory allocation of ... bytes failed`
* Linux with `vm.overcommit_memory = 1` results in `killed`
* Linux with `vm.overcommit_memory = 2` results in `memory allocation of ... bytes failed`

MacOS always works without exiting.

Maybe there's not much we can do here. But doing nothing will not adere the specification (i.e. _The [table.grow](https://webassembly.github.io/spec/core/bikeshed/index.html#syntax-instr-table) instruction grows table by a given delta and returns the previous size, or −1 if enough space cannot be allocated_).